src/web/forms/fields/AutoCalc/index.js

```diff
@@ -20,20 +20,79 @@ import React from 'react';
 import { connect } from 'react-redux';
 import _get from 'lodash.get';
 import functions from './functions';
+import classNames from 'classnames';
+
+const alertFaIconClasses = {
+  danger: 'warning',
+  warning: 'warning',
+  success: 'check',
+};
 
 export const AutoCalcComponent = ({
   value,
   style,
   precision,
+  alerts,
+  type,
 }: {
-  value: string,
+  value?: string,
   style: ?Object,
   precision: ?number,
-}) => (
-  <span className="form-static" style={style}>
-    {(precision && typeof value === 'number') ? value.toFixed(precision) : value}
-  </span>
-);
+  type?: string,
+  alerts?: Array<Object>,
+}) => {
+  let alert = null;
+  const overrideStyle = {};
+  if (type === 'number' && alerts) {
+    const numValue = parseFloat(value);
+
+    alert = alerts.find(al =>
+        ((al.range[0] == null || numValue >= al.range[0])
+          && (al.range[1] == null || al.range[1] > numValue))
+      );
+
+    if (style && style.width) {
+      overrideStyle.width = style.width + 32;
+    }
+  }
+
+  return (
+    <span className="form-static" style={style}>
+      <table>
```

Comment:
Using table just for layout is bad practice in modern web.
https://stackoverflow.com/questions/83073/why-not-use-tables-for-layout-in-html/84986#84986

Additionally, specifying layout parameter such as `<td width="389">` is likely to break the layout sometime.

Layout must be done by CSS.